utils/system-notification/files/usr/libexec/rpcd/system.notification

```diff
@@ -0,0 +1,190 @@
+#!/bin/sh
+#
+# Copyright (C) 2021 TDT AG <development@tdt.de>
+#
+# This is free software, licensed under the Apache License Version 2.0.
+# See https://www.apache.org/licenses/LICENSE-2.0 for more information.
+#
+
+. /usr/share/libubox/jshn.sh
+
+MESSAGE_DIR="/tmp/notification/messages.d"
+
+get_uptime() {
+	local uptime
+
+	uptime=$(cat /proc/uptime)
+	echo "${uptime%%.*}"
+}
+
+get_id() {
+	local file="$1"
+	local id
+
+	id="${file##*/}"
+
+	echo "${id%*.json}"
+}
+
+list() {
+	json_init
+
+	json_add_object "add"
+	json_add_string "service"
+	json_add_string "title"
+	json_add_string "message"
+	json_close_object
+
+	json_add_object "remove"
+	json_add_int "number"
+	json_close_object
+
+	json_add_object "list"
+	json_close_object
+
+	json_add_object "flush"
+	json_close_object
+
+	json_dump
+}
+
+notification_add() {
+	local service="$1"
+	local title="$2"
+	local message="$3"
+
+	local timestamp uptime
+
+	local code=-1
+	local msg="Unable to add Notification"
+
+	timestamp="$(date +%s)"
+	uptime="$(get_uptime)"
+
+	json_init
+	json_add_string service "$service"
+	json_add_string title "$title"
+	json_add_string message "$message"
+	json_add_int timestamp "$timestamp"
+	json_dump > "${MESSAGE_DIR}/${uptime}.json"
```

Comment:
what's the purpose of the uptime here? is it just to get a "unique" filename to use?  what was wrong with the timestamp you already have? your uptime call throws away fractional seconds anyway, so it seems to be just same same, but different.  You  and you can use $$ to add the current process pid as well, or even just use mktemp?  it seems to be available in the default busybox configs.

Reply:
> what's the purpose of the uptime here? is it just to get a "unique" filename to use? what was wrong with the timestamp you already have? your uptime call throws away fractional seconds anyway, so it seems to be just same same, but different.

I have a target without RTC so when there is an ntp connection the time is set during the runtime.
If I use the uptime it continues to be counted forward and does not jump during an ntp synchronization.

But I think your suggestion with the **mktemp** is not bad, I think I will rewrite this :+1:

Reply:
@karlp I just realised, or rather I remembered, that mktemp generates random alphanumeric file names
But I wanted only use a number for the id.
It should be an uint and not a string.
I guess mktemp doesn't work for this case.

I would use the timestamp this has the advantage that I could write more then one message per second and it is a unique id that increments.

Reply:
it doesn't matter if it jumps, you're just this is a fairly poor unique id :)  you should perhaps just use a uuid or something anyway.  perhaps, cat /proc/sys/kernel/random/uuid and never have to consider this anymore.... ?

Reply:
(and why can't it be a string, there's no significance to the number, and I don't really buy that this is a json optimization for  ints vs strings, the "benefit" is far too small for something that is intended as notifications.  you're not generating thousands of these per minute, let alone per second....

Reply:
Thanks for the good advice I have now selected the id of the message as the uuid string.